src/node_api.cc

```diff
@@ -706,6 +706,40 @@ bool FindWrapper(v8::Local<v8::Object> obj,
   return true;
 }
 
+static void DeleteEnv(const v8::WeakCallbackInfo<napi_env__>& data) {
+  delete data.GetParameter();
+}
+
+napi_env get_env(v8::Local<v8::Context> context) {
+  napi_env result;
+
+  auto isolate = context->GetIsolate();
+  auto global = context->Global();
+
+  auto maybe_key = v8::String::NewFromUtf8(isolate, "N-API Environment",
+      v8::NewStringType::kInternalized);
+  CHECK(!maybe_key.IsEmpty());
+  auto key = v8::Private::ForApi(isolate, maybe_key.ToLocalChecked());
```

Comment:
I thought the point of caching the `v8::Private` was to avoid calling `ForApi` more than once per new isolate. Or did you mean that I should cache the result of `v8::String::NewFromOneByte`?

If I use a `std::map` I must protect it with a mutex because another thread might be in the process of depositing a new value or it might be in the process of erasing an unused value.

I suppose you mean the kind of caching that I see in node.cc, like `env->exports_string()`. I guess maybe I should start looking into using `node::Environment`.